### ISSUE TITLE
vite + react: PURE annotation that Rollup cannot interpret

### ISSUE THREAD

**Issue by BPreisner** [Anber/wyw-in-js#74]
## Packages:

- wyw-in-js version: **0.5.1**
- Bundler: **vite@5.2.8**
- **@linaria/react@6.1.0**

## Description
There is an issue with simple `styled` interpolation upon vite build.

![image](https://github.com/Anber/wyw-in-js/assets/28753683/37c07e78-5177-4497-9f11-ac5345db4f17)

When building with Vite:

![image](https://github.com/Anber/wyw-in-js/assets/28753683/44c41de6-3cbd-48f7-b557-e149fcbc29aa)

I was trying to switch from styled-components to linaria for our components library and this error is populated for **every** interpolation in the project. I worry that it's not just a polite warning and may lead to breaking something later.

## Reproducible Demo

https://stackblitz.com/edit/vitejs-vite-gfwqu5?file=src%2FApp.tsx

`$ npm run build`

(There is no issue when running `dev` command)

**Comment by kriakiku** [comment]
Same. With rollup:

```
{
  "@wyw-in-js/rollup": "0.5.4",
  "@linaria/react": "6.2.1",
  "rollup": "4.24.0"
}
```

Example of how JS differs after bundling (Left is version 4 of linaria, right is version 6):
![image](https://github.com/user-attachments/assets/97a1cc81-1bec-4135-a673-608716de0298)

**Comment by agriffis** [comment]
I tried to use patch-package to change these to `/*#__NO_SIDE_EFFECTS__*/` but still got a bunch of complaints from rollup. Has anybody else tried this?

**Comment by aeharding** [comment]
> I tried to use patch-package to change these to /*#__NO_SIDE_EFFECTS__*/ but still got a bunch of complaints from rollup. Has anybody else tried this?

No... but it seems harmless enough. I am using Rollup's 'onwarn' option to ignore the warnings for this.

**Comment by agriffis** [comment]
> No... but it seems harmless enough. I am using Rollup's 'onwarn' option to ignore the warnings for this.

Sure, you can ignore the warnings. I was hoping to enable the optimization that the pragma intends to achieve, which the warnings indicate is not working.
